optout/mock_saver.go

```diff
@@ -0,0 +1,22 @@
+package optout
+
+type MockSaver struct {
+	Files         []OptOutFile
+	OptOutRecords []OptOutRecord
+}
+
+func (m *MockSaver) SaveFile(optOutFile OptOutFile) (fileID uint, err error) {
```

Comment:
testify has a tool for mocking that we use elsewhere in our tests; out of curiosity, was it not a good fit here?

Reply:
Good question! I think the difference here is to have something that can store state, whereas the testify mocking is more about mocking specific calls.

I think I could probably have used testify and done the test assertions differently (checking specific calls to "UpdateImportStatus" for example, instead of checking the internal end state) but it's a little more awkward to read and write.

This post is behind a paywall, bu the first part seems to align with my thinking about testing state vs. testing calls 🙂 I think what I'm making here is not really a "mock" saver but a "fake" in-memory saver.

https://medium.com/@june.pravin/mocking-is-not-practical-use-fakes-e30cc6eaaf4e

Reply:
ah, that makes much more sense; fake vs mock. Not required, but maybe a name update would provide more clarity? Ex: `FakeSaver` vs `MockSaver`